src/Sentry/Envelopes/Envelope.cs

```diff
@@ -180,7 +180,18 @@ internal void Serialize(Stream stream, IDiagnosticLogger? logger, ISystemClock c
                 {
                     try
                     {
-                        items.Add(EnvelopeItem.FromAttachment(attachment));
+                        // We pull the stream out here so we can length check
+                        // to avoid adding an invalid attachment
+                        var stream = attachment.Content.GetStream();
+                        if (stream.Length != 0)
+                        {
+                            items.Add(EnvelopeItem.FromAttachment(attachment, stream));
+                        }
+                        else
+                        {
+                            logger?.LogWarning("Did not add '{0}' to envelope because the stream was empty.",
```

Comment:
For streams that are not buffered, could this return `0` before you actually do the first `Read`?

Reply:
That's a really good point.  To address that, we may have to buffer the stream here, which feels wrong...  Thinking...

Reply:
Or we can just type check.  Feels like cheating, but will work

Reply:
Oh, we already have an extension for this, we can just use that.

Reply:
FYI - We double checked a non-buffered stream.  As long as its seekable, the length will return correctly.  Specifically, a `FileStream` will give the length of the file even if you haven't read it yet - which is probably what people will be using the most on this API.